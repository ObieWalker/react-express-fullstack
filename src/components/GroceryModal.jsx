import React from 'react';
import { Modal } from 'react-bootstrap'

const GroceryModal = ({errors, addGrocery, editGrocery, updateDetails, show, onHide, groceryName, groceryPrice, grocery}) => {
  return (
    <div>
      <Modal
          show={show}
          onHide={onHide}
          aria-labelledby="contained-modal-title"
          >
          <Modal.Header closeButton>
            <Modal.Title id="contained-modal-title">Grocery Details
            </Modal.Title>
            </Modal.Header>
            <Modal.Body style={{ width: '100%', marginLeft: 'auto', marginRight: 'auto'}}>
            <label htmlFor="name" className="active">
              Grocery Name:
            </label>
            {errors.groceryName && (
              <span className="alert-danger"><br />
                {errors.groceryName}
              </span>
            )}
            <input
              type="text"
              className="form-control"
              defaultValue={groceryName || ''}
              id="groceryName"
              name="groceryName"
              required
              onChange={updateDetails}
            />
            <br />
            <label htmlFor="name" className="active">
              Grocery Price (&#8358;):
            </label>
            {errors.groceryPrice && (
              <span className="alert-danger"><br />
                {errors.groceryPrice}
              </span>
            )}
            <input
              type="number"
              className="form-control"
              defaultValue={groceryPrice || ''}
              id="groceryPrice"
              name="groceryPrice"
              required
              onChange={updateDetails}
            />
            </Modal.Body>
            <Modal.Footer>
            {grocery._id ? 
              <button
              type="submit"
              id="add-grocery"
              className="waves-effect waves-dark
              btn right hoverable dark-green btn btn-primary"
              onClick={() => editGrocery(grocery)}
              >Edit Item
            </button>
            : <button
              type="submit"
              id="add-grocery"
              className="waves-effect waves-dark
              btn right hoverable dark-green btn btn-primary"
              onClick={addGrocery}
              >Add Item
            </button>}
            
            </Modal.Footer>
          </Modal>
    </div>
  );
}

export default GroceryModal;
