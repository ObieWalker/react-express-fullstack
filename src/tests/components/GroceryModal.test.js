import React from 'react';
import ReactDOM from 'react-dom';
import expect from 'expect';
import { shallow } from 'enzyme';
import GroceryModal from '../../components/GroceryModal';

let mounted;
let props;
let wrapper;
const groceryName = ''
const groceryPrice = ''
const errors = {
  groceryName : '',
  groceryPrice: ''
}
const addGrocery = jest.fn();
const editGrocery = jest.fn();
const updateDetails = jest.fn();
const show = true
const onHide = jest.fn();
const  grocery = {}

const getComponent = () => {
  if (!mounted) {
    props = {
      errors,
      addGrocery,
      editGrocery,
      updateDetails,
      show,
      onHide,
      groceryName,
      groceryPrice,
      grocery
    };
    mounted = shallow(<GroceryModal {...props} />);
  }
  return mounted;
};

describe('GroceryModal Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

});

