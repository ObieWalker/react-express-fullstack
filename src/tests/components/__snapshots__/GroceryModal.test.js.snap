// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroceryModal Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroceryModal
    addGrocery={[MockFunction]}
    editGrocery={[MockFunction]}
    errors={
      Object {
        "groceryName": "",
        "groceryPrice": "",
      }
    }
    grocery={Object {}}
    groceryName=""
    groceryPrice=""
    onHide={[MockFunction]}
    show={true}
    updateDetails={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        animation={true}
        aria-labelledby="contained-modal-title"
        autoFocus={true}
        backdrop={true}
        bsClass="modal"
        dialogComponentClass={[Function]}
        enforceFocus={true}
        keyboard={true}
        manager={
          ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          }
        }
        onHide={[MockFunction]}
        renderBackdrop={[Function]}
        restoreFocus={true}
        show={true}
      >
        <ModalHeader
          bsClass="modal-header"
          closeButton={true}
          closeLabel="Close"
        >
          <ModalTitle
            bsClass="modal-title"
            componentClass="h4"
            id="contained-modal-title"
          >
            Grocery Details
          </ModalTitle>
        </ModalHeader>
        <ModalBody
          bsClass="modal-body"
          componentClass="div"
          style={
            Object {
              "marginLeft": "auto",
              "marginRight": "auto",
              "width": "100%",
            }
          }
        >
          <label
            className="active"
            htmlFor="name"
          >
            Grocery Name:
          </label>
          
          <input
            className="form-control"
            defaultValue=""
            id="groceryName"
            name="groceryName"
            onChange={[MockFunction]}
            required={true}
            type="text"
          />
          <br />
          <label
            className="active"
            htmlFor="name"
          >
            Grocery Price (₦):
          </label>
          
          <input
            className="form-control"
            defaultValue=""
            id="groceryPrice"
            name="groceryPrice"
            onChange={[MockFunction]}
            required={true}
            type="number"
          />
        </ModalBody>
        <ModalFooter
          bsClass="modal-footer"
          componentClass="div"
        >
          <button
            className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
            id="add-grocery"
            onClick={[MockFunction]}
            type="submit"
          >
            Add Item
          </button>
        </ModalFooter>
      </Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": true,
        "aria-labelledby": "contained-modal-title",
        "autoFocus": true,
        "backdrop": true,
        "bsClass": "modal",
        "children": Array [
          <ModalHeader
            bsClass="modal-header"
            closeButton={true}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title"
            >
              Grocery Details
            </ModalTitle>
          </ModalHeader>,
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
            style={
              Object {
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "100%",
              }
            }
          >
            <label
              className="active"
              htmlFor="name"
            >
              Grocery Name:
            </label>
            
            <input
              className="form-control"
              defaultValue=""
              id="groceryName"
              name="groceryName"
              onChange={[MockFunction]}
              required={true}
              type="text"
            />
            <br />
            <label
              className="active"
              htmlFor="name"
            >
              Grocery Price (₦):
            </label>
            
            <input
              className="form-control"
              defaultValue=""
              id="groceryPrice"
              name="groceryPrice"
              onChange={[MockFunction]}
              required={true}
              type="number"
            />
          </ModalBody>,
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <button
              className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
              id="add-grocery"
              onClick={[MockFunction]}
              type="submit"
            >
              Add Item
            </button>
          </ModalFooter>,
        ],
        "dialogComponentClass": [Function],
        "enforceFocus": true,
        "keyboard": true,
        "manager": ModalManager {
          "add": [Function],
          "containers": Array [],
          "data": Array [],
          "handleContainerOverflow": true,
          "hideSiblingNodes": true,
          "isTopModal": [Function],
          "modals": Array [],
          "remove": [Function],
        },
        "onHide": [MockFunction],
        "renderBackdrop": [Function],
        "restoreFocus": true,
        "show": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-header",
            "children": <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title"
            >
              Grocery Details
            </ModalTitle>,
            "closeButton": true,
            "closeLabel": "Close",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-title",
              "children": "Grocery Details",
              "componentClass": "h4",
              "id": "contained-modal-title",
            },
            "ref": null,
            "rendered": "Grocery Details",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-body",
            "children": Array [
              <label
                className="active"
                htmlFor="name"
              >
                Grocery Name:
              </label>,
              "",
              <input
                className="form-control"
                defaultValue=""
                id="groceryName"
                name="groceryName"
                onChange={[MockFunction]}
                required={true}
                type="text"
              />,
              <br />,
              <label
                className="active"
                htmlFor="name"
              >
                Grocery Price (₦):
              </label>,
              "",
              <input
                className="form-control"
                defaultValue=""
                id="groceryPrice"
                name="groceryPrice"
                onChange={[MockFunction]}
                required={true}
                type="number"
              />,
            ],
            "componentClass": "div",
            "style": Object {
              "marginLeft": "auto",
              "marginRight": "auto",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Grocery Name:",
                "className": "active",
                "htmlFor": "name",
              },
              "ref": null,
              "rendered": "Grocery Name:",
              "type": "label",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "defaultValue": "",
                "id": "groceryName",
                "name": "groceryName",
                "onChange": [MockFunction],
                "required": true,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Grocery Price (₦):",
                "className": "active",
                "htmlFor": "name",
              },
              "ref": null,
              "rendered": "Grocery Price (₦):",
              "type": "label",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "defaultValue": "",
                "id": "groceryPrice",
                "name": "groceryPrice",
                "onChange": [MockFunction],
                "required": true,
                "type": "number",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-footer",
            "children": <button
              className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
              id="add-grocery"
              onClick={[MockFunction]}
              type="submit"
            >
              Add Item
            </button>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Item",
              "className": "waves-effect waves-dark btn right hoverable dark-green btn btn-primary",
              "id": "add-grocery",
              "onClick": [MockFunction],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add Item",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          animation={true}
          aria-labelledby="contained-modal-title"
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[MockFunction]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={true}
        >
          <ModalHeader
            bsClass="modal-header"
            closeButton={true}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title"
            >
              Grocery Details
            </ModalTitle>
          </ModalHeader>
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
            style={
              Object {
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "100%",
              }
            }
          >
            <label
              className="active"
              htmlFor="name"
            >
              Grocery Name:
            </label>
            
            <input
              className="form-control"
              defaultValue=""
              id="groceryName"
              name="groceryName"
              onChange={[MockFunction]}
              required={true}
              type="text"
            />
            <br />
            <label
              className="active"
              htmlFor="name"
            >
              Grocery Price (₦):
            </label>
            
            <input
              className="form-control"
              defaultValue=""
              id="groceryPrice"
              name="groceryPrice"
              onChange={[MockFunction]}
              required={true}
              type="number"
            />
          </ModalBody>
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <button
              className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
              id="add-grocery"
              onClick={[MockFunction]}
              type="submit"
            >
              Add Item
            </button>
          </ModalFooter>
        </Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": true,
          "aria-labelledby": "contained-modal-title",
          "autoFocus": true,
          "backdrop": true,
          "bsClass": "modal",
          "children": Array [
            <ModalHeader
              bsClass="modal-header"
              closeButton={true}
              closeLabel="Close"
            >
              <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
                id="contained-modal-title"
              >
                Grocery Details
              </ModalTitle>
            </ModalHeader>,
            <ModalBody
              bsClass="modal-body"
              componentClass="div"
              style={
                Object {
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "100%",
                }
              }
            >
              <label
                className="active"
                htmlFor="name"
              >
                Grocery Name:
              </label>
              
              <input
                className="form-control"
                defaultValue=""
                id="groceryName"
                name="groceryName"
                onChange={[MockFunction]}
                required={true}
                type="text"
              />
              <br />
              <label
                className="active"
                htmlFor="name"
              >
                Grocery Price (₦):
              </label>
              
              <input
                className="form-control"
                defaultValue=""
                id="groceryPrice"
                name="groceryPrice"
                onChange={[MockFunction]}
                required={true}
                type="number"
              />
            </ModalBody>,
            <ModalFooter
              bsClass="modal-footer"
              componentClass="div"
            >
              <button
                className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
                id="add-grocery"
                onClick={[MockFunction]}
                type="submit"
              >
                Add Item
              </button>
            </ModalFooter>,
          ],
          "dialogComponentClass": [Function],
          "enforceFocus": true,
          "keyboard": true,
          "manager": ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          },
          "onHide": [MockFunction],
          "renderBackdrop": [Function],
          "restoreFocus": true,
          "show": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-header",
              "children": <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
                id="contained-modal-title"
              >
                Grocery Details
              </ModalTitle>,
              "closeButton": true,
              "closeLabel": "Close",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-title",
                "children": "Grocery Details",
                "componentClass": "h4",
                "id": "contained-modal-title",
              },
              "ref": null,
              "rendered": "Grocery Details",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-body",
              "children": Array [
                <label
                  className="active"
                  htmlFor="name"
                >
                  Grocery Name:
                </label>,
                "",
                <input
                  className="form-control"
                  defaultValue=""
                  id="groceryName"
                  name="groceryName"
                  onChange={[MockFunction]}
                  required={true}
                  type="text"
                />,
                <br />,
                <label
                  className="active"
                  htmlFor="name"
                >
                  Grocery Price (₦):
                </label>,
                "",
                <input
                  className="form-control"
                  defaultValue=""
                  id="groceryPrice"
                  name="groceryPrice"
                  onChange={[MockFunction]}
                  required={true}
                  type="number"
                />,
              ],
              "componentClass": "div",
              "style": Object {
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Grocery Name:",
                  "className": "active",
                  "htmlFor": "name",
                },
                "ref": null,
                "rendered": "Grocery Name:",
                "type": "label",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-control",
                  "defaultValue": "",
                  "id": "groceryName",
                  "name": "groceryName",
                  "onChange": [MockFunction],
                  "required": true,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Grocery Price (₦):",
                  "className": "active",
                  "htmlFor": "name",
                },
                "ref": null,
                "rendered": "Grocery Price (₦):",
                "type": "label",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-control",
                  "defaultValue": "",
                  "id": "groceryPrice",
                  "name": "groceryPrice",
                  "onChange": [MockFunction],
                  "required": true,
                  "type": "number",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-footer",
              "children": <button
                className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
                id="add-grocery"
                onClick={[MockFunction]}
                type="submit"
              >
                Add Item
              </button>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Item",
                "className": "waves-effect waves-dark btn right hoverable dark-green btn btn-primary",
                "id": "add-grocery",
                "onClick": [MockFunction],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Add Item",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
